app/src/main/java/com/eveningoutpost/dexdrip/ShareModels/ShareRest.java

```diff
@@ -130,19 +153,133 @@ public boolean verify(String hostname, SSLSession session) {
 
             return okHttpClient;
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            throw new RuntimeException("Error occurred initializing OkHttp: ", e);
         }
     }
 
-    public static OkClient getOkClient (){
-        OkHttpClient client1 = getOkHttpClient();
-        OkClient _client = new OkClient(client1);
-        return _client;
+    public String getSessionId() {
+        AsyncTask<String, Void, String> task = new AsyncTask<String, Void, String>() {
+
+            @Override
+            protected String doInBackground(String... params) {
+                try {
+                    Boolean isActive = null;
+                    if (params[0] != null)
+                        isActive = dexcomShareApi.checkSessionActive(params[0]).execute().body();
+                    if (isActive == null || !isActive) {
+                        return updateAuthenticationParams();
+                    } else
+                        return params[0];
+                } catch (IOException e) {
+                    return null;
+                }
+            }
+
+            private String updateAuthenticationParams() throws IOException {
+                sessionId = dexcomShareApi.getSessionId(new ShareAuthenticationBody(password, username).toMap()).execute().body();
+                dexcomShareApi.authenticatePublisherAccount(sessionId, serialNumber, new ShareAuthenticationBody(password, username).toMap()).execute().body();
+                dexcomShareApi.StartRemoteMonitoringSession(sessionId, serialNumber).execute();
+                String assignment = dexcomShareApi.checkMonitorAssignment(sessionId, serialNumber).execute().body();
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1306138860/issues/source?bid=2735053&fileBranchId=2947762#l183)